### PR TITLE
Frontend: update the user guide

### PR DESCRIPTION
**Pull request by ongspxm**
@eugenepeh the user guide is ready to be merged. are these the changes you are looking for?

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -18,8 +18,12 @@
    * Alternatively, you can compile the executable Jar yourself by following our [build from source guide](Build.md).
 2. Execute it on the OS terminal. <br>
 Usage: `java -jar RepoSense.jar -config CSV_CONFIG_FILE_PATH [-output OUTPUT_DIRECTORY] [-since DD/MM/YYYY] [-until DD/MM/YYYY]`
-
-3. The dashboard will be generated in the designated OUTPUT_DIRECTORY, or current working directory otherwise, upon successful compilation as index.html.
+3. The dashboard will be generated in the designated OUTPUT_DIRECTORY, or current working directory otherwise, upon successful compilation as `index.html`.
+4. Start a webserver in the generated report directory to server the files
```
Comment (eugenepeh) [inline docs/UserGuide.md]: please provide a clearer explanation. 
E.g. 
To visualize the report... 

Alternatively...

File: docs/UserGuide.md [hunk 2]
```diff
@@ -19,11 +19,12 @@
...
 Usage: `java -jar RepoSense.jar -config CSV_CONFIG_FILE_PATH [-output OUTPUT_DIRECTORY] [-since DD/MM/YYYY] [-until DD/MM/YYYY]`
 3. The dashboard will be generated in the designated OUTPUT_DIRECTORY, or current working directory otherwise, upon successful compilation as `index.html`.
-4. Start a webserver in the generated report directory to server the files
+4. To visualize the report, start a webserver in the generated report directory to server the files
     - [python](http://www.pythonforbeginners.com/modules-in-python/how-to-use-simplehttpserver/)
     - [darkhttpd](https://unix4lyfe.org/darkhttpd/)
     - [apache httpd](https://httpd.apache.org/download.cgi)
-5. If unable to complete step 4, you can also open the `index.html` locally and select the zipped file to load the content in.
+5. The dashboard will be available on the port opened by your web server of choice.
+5. Alterntively, if unable to complete step 4, you can also open the `index.html` locally and select the zipped file to load the content in.
```
Comment (eugenepeh) [inline docs/UserGuide.md]: Alterntively -> `Alternatively`

Comment (eugenepeh) [reply docs/UserGuide.md]: This can be 4a. instead

File: docs/UserGuide.md [hunk 3]
```diff
@@ -19,11 +19,12 @@
 2. Execute it on the OS terminal. <br>
 Usage: `java -jar RepoSense.jar -config CSV_CONFIG_FILE_PATH [-output OUTPUT_DIRECTORY] [-since DD/MM/YYYY] [-until DD/MM/YYYY]`
 3. The dashboard will be generated in the designated OUTPUT_DIRECTORY, or current working directory otherwise, upon successful compilation as `index.html`.
-4. Start a webserver in the generated report directory to server the files
+4. To visualize the report, start a webserver in the generated report directory to server the files
     - [python](http://www.pythonforbeginners.com/modules-in-python/how-to-use-simplehttpserver/)
     - [darkhttpd](https://unix4lyfe.org/darkhttpd/)
     - [apache httpd](https://httpd.apache.org/download.cgi)
-5. If unable to complete step 4, you can also open the `index.html` locally and select the zipped file to load the content in.
+5. The dashboard will be available on the port opened by your web server of choice.
```
Comment (eugenepeh) [inline docs/UserGuide.md]: Also, there is a double count, perhaps this should be listed as point form under 4.

File: docs/UserGuide.md [hunk 4]
```diff
@@ -19,11 +19,12 @@
 2. Execute it on the OS terminal. <br>
 Usage: `java -jar RepoSense.jar -config CSV_CONFIG_FILE_PATH [-output OUTPUT_DIRECTORY] [-since DD/MM/YYYY] [-until DD/MM/YYYY]`
 3. The dashboard will be generated in the designated OUTPUT_DIRECTORY, or current working directory otherwise, upon successful compilation as `index.html`.
-4. Start a webserver in the generated report directory to server the files
+4. To visualize the report, start a webserver in the generated report directory to server the files
```
Comment (eugenepeh) [inline docs/UserGuide.md]: Doesn't really explain well. Try to be more elaborative, e.g. why is it needed to start a webserver?

Alternatively, I think we should just scrape the entire web-server thing for local preview. People are unlikely to use it unless you're able to deliver them the cost and benefit. Let's go with using zipped file for local instead.

File: docs/UserGuide.md (hunk 2 above)
Comment (ongspxm) [reply docs/UserGuide.md]: oh, paiseh, the double 5 was a mistake.

unable to do 4a, won't render as 4a anyway.

> It’s important to note that the actual numbers you use to mark the list have no effect on the HTML output Markdown produces.
[source](https://daringfireball.net/projects/markdown/syntax#list)

File: docs/UserGuide.md (hunk 4 above)
Comment (ongspxm) [reply docs/UserGuide.md]: roger that, removing the web server part

**Comment by ongspxm** [comment]
@AdityaA1998 @yamidark @yong24s review docs pls

File: docs/UserGuide.md [hunk 5]
```diff
@@ -18,8 +18,8 @@
    * Alternatively, you can compile the executable Jar yourself by following our [build from source guide](Build.md).
 2. Execute it on the OS terminal. <br>
 Usage: `java -jar RepoSense.jar -config CSV_CONFIG_FILE_PATH [-output OUTPUT_DIRECTORY] [-since DD/MM/YYYY] [-until DD/MM/YYYY]`
-
-3. The dashboard will be generated in the designated OUTPUT_DIRECTORY, or current working directory otherwise, upon successful compilation as index.html.
+3. The dashboard will be generated in the designated OUTPUT_DIRECTORY, or current working directory otherwise, upon successful compilation as `index.html`.
+4. Open the `index.html` for the dashboard, and upload `archive.zip` to start loading the data.
```
[the remark this answers is not in the document]
Comment (ongspxm) [reply docs/UserGuide.md]: roger that,

File: docs/UserGuide.md [hunk 6]
```diff
@@ -2,24 +2,22 @@
...
 
 ## How to Generate Dashboard
 1. Download the latest executable Jar on our [release](https://github.com/reposense/RepoSense/releases/latest).
    * Alternatively, you can compile the executable Jar yourself by following our [build from source guide](Build.md).
-2. Execute it on the OS terminal. <br>
+1. Execute it on the OS terminal. <br>
 Usage: `java -jar RepoSense.jar -config CSV_CONFIG_FILE_PATH [-output OUTPUT_DIRECTORY] [-since DD/MM/YYYY] [-until DD/MM/YYYY]`
-3. The dashboard will be generated in the designated OUTPUT_DIRECTORY, or current working directory otherwise, upon successful compilation as `index.html`.
-4. Open the `index.html` for the dashboard, and upload `archive.zip` to start loading the data.
+1. The dashboard will be generated in the designated OUTPUT_DIRECTORY, or current working directory otherwise, upon successful compilation as `index.html`.
```
Comment (eugenepeh) [inline docs/UserGuide.md]: The ~dashboard~ `report`

File: docs/UserGuide.md [hunk 7]
```diff
@@ -2,24 +2,22 @@
...
+1. Open the `index.html` for the dashboard, and upload `archive.zip` to start loading the data.
```
Comment (eugenepeh) [inline docs/UserGuide.md]: To visualize the report, open the `index.html`. If the dashboard is not loaded automatically, upload the `archive.zip` (generated in the output directory) manually to load the data.

File: docs/UserGuide.md [hunk 8]
```diff
@@ -2,24 +2,22 @@
...
 ## How to Generate Dashboard
 1. Download the latest executable Jar on our [release](https://github.com/reposense/RepoSense/releases/latest).
    * Alternatively, you can compile the executable Jar yourself by following our [build from source guide](Build.md).
-2. Execute it on the OS terminal. <br>
+1. Execute it on the OS terminal. <br>
 Usage: `java -jar RepoSense.jar -config CSV_CONFIG_FILE_PATH [-output OUTPUT_DIRECTORY] [-since DD/MM/YYYY] [-until DD/MM/YYYY]`
-
-3. The dashboard will be generated in the designated OUTPUT_DIRECTORY, or current working directory otherwise, upon successful compilation as index.html.
+1. The dashboard will be generated in the designated OUTPUT_DIRECTORY, or current working directory otherwise, upon successful compilation as `index.html`.
+1. Open the `index.html` for the dashboard, and upload `archive.zip` to start loading the data.
```
Comment (yong24s) [inline docs/UserGuide.md]: Open the `index.html`, and upload `archive.zip` from the OUTPUT_DIRECTORY to view the dashboard.

Comment (yong24s) [reply docs/UserGuide.md]: Take Eugene's suggestion instead.

**Comment by ongspxm** [comment]
@eugenepeh ready for review

**Pull request merged**